src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/VS2kAddItemCommandHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Input;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    [ExportCommandGroup(CommandGroup.VisualStudioStandard2k)]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class VS2kAddItemCommandHandler : AbstractAddItemCommandHandler
+    {
+        protected override Dictionary<long, List<TemplateDetails>> GetTemplateDetails() => new Dictionary<long, List<TemplateDetails>>
```

Comment:
Returns a new value every time.

Reply:
Understand that this is cached, but we basically have a representation of this object per configuration per project, when it never changes. Store a stack field, throw away the lazy and just return the static field.